pkg/services/accesscontrol/database/resource_permissions.go

```diff
@@ -87,6 +105,12 @@ func (s *AccessControlStore) SetBuiltInResourcePermission(ctx context.Context, o
 
 	err = s.sql.WithTransactionalDbSession(ctx, func(sess *sqlstore.DBSession) error {
 		permission, err = s.setResourcePermission(sess, orgID, managedBuiltInRoleName(builtInRole), s.builtInRoleAdder(sess, orgID, builtInRole), cmd)
+		if err != nil {
+			return err
+		}
+		if hook != nil {
+			return hook(sess, orgID, builtInRole, cmd.ResourceID, cmd.Permission)
+		}
 		return err
```

Comment:
shouldn't this be specifically return nil?

```suggestion
		return nil
```

Reply:
or 

```go
		if err == nil && hook != nil {
			return hook(sess, orgID, builtInRole, cmd.ResourceID, cmd.Permission)
		}
		
		return err
```

Reply:
should match L78-84 at least :)